pytensor/tensor/math.py

```diff
@@ -343,6 +345,208 @@ def grad(self, inp, grads):
         return (g_x,)
 
 
+class TensorMax(COp):
```

Comment:
We already have a Max Op implemented, we can just use that. Only thing is we need to port the gradient methods (and any other methods that MaxAndArgmax had but Max didn't)

Reply:
I have added the grad methods locally and it works. MaxandArgmax additionally supports ```c_code``` and ```R_op``` methods. Do you suggest porting them to? 

The c_code is giving me problem as:
https://github.com/pymc-devs/pytensor/blob/eb18f0ea4d7cc5592149f50780c4d3f6c021e90c/pytensor/tensor/math.py#L194-L200

The ```sub``` that I get using Max is:
```
self = Max(scalar_op=maximum,axis=(0,),dtype=None,acc_dtype=None,upcast_discrete_output=False)
node = Max{axis=0}([[ 5.6641e ... 6438e+01]])
name = 'node_<<<<HASH_PLACEHOLDER>>>>_0', inp = ['V3'], out = ['V1']
sub = {'fail': '{\n        __failure = 5;\n        if (!PyErr_Occurred()) {\n            PyErr_SetString(PyExc_RuntimeError,...           "No Python exception was set.");\n        }\n        goto __label_5;}', 'failure_var': '__failure', 'id': 5}
```
Hence I run into an error:
```
>       axis = sub["params"]
E       KeyError: 'params'
```

Reply:
Max has C code by default as being a subclass of CAReduce.

Argmax would be useful to have C implementation... we can port the respective part from before.